API/fleece/Fleece.hh

```diff
@@ -326,9 +326,8 @@ namespace fleece {
             FLSharedKeys FL_NULLABLE sk =nullptr,
             slice externDest =nullslice) noexcept
         {
-            auto sliceResult = FLSliceResult(std::move(fleeceData));
+            FLSliceResult sliceResult {fleeceData.buf, fleeceData.size};
```

Comment:
This probably deserves a comment on why it is doing things like this (i.e. to avoid over-retaining).